### PR TITLE
Destroy Env script enhancements

### PR DESCRIPTION
**Pull request by oren-zohar**
### Summary of your changes
Small `Destroy Env` script changes:
- Linting
- Ensure it exits with an error if one of the Envs failed to delete

### CODE REVIEW

File: deploy/test-environments/delete_env.sh
Comment (orestisfl) [inline deploy/test-environments/delete_env.sh:27]:
```suggestion
    if aws s3 cp "$BUCKET/$ENV/terraform.tfstate" "$tfstate"; then
```

Slightly more readable IMO

File: deploy/test-environments/delete_env.sh
Comment (orestisfl) [inline deploy/test-environments/delete_env.sh:34]:
```suggestion
            aws s3 rm "$BUCKET/$ENV" --recursive; then
```

**Review by orestisfl — COMMENTED** [review]
While you are at it, can you also add the script in the pre-commit hook for shellcheck?

**Review by gurevichdmitry — APPROVED** [review]
@oren-zohar, LGTM

**Review by orestisfl — APPROVED** [review]

**Comment by oren-zohar** [comment]
I'll run it on the branch before merging, please don't merge

File: .pre-commit-config.yaml
Comment (orestisfl) [inline .pre-commit-config.yaml:80]:
```suggestion
        files: (deploy/.*sh|scripts/publish_cft.sh|.ci/scripts/.*sh)
        exclude: deploy/deployment-manager/.*sh
```

**Pull request merged**
